<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">@import url(https://themes.googleusercontent.com/fonts/css?kit=fpjTOVmNbO4Lz34iLyptLROjXfSpIU_1TsQlCJ7q4WkfnUSgpG7o6Yx3LgtCQhIzTaYikgKmitzicuqVhRTMPFrAsqVpAzQBvtVDn-47_eg);.lst-kix_ejwho1vu4d53-7>li:before{content:"-  "}.lst-kix_ejwho1vu4d53-8>li:before{content:"-  "}ol.lst-kix_rvwmfr82j1qn-4.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-4 0}.lst-kix_ejwho1vu4d53-5>li:before{content:"-  "}.lst-kix_ejwho1vu4d53-2>li:before{content:"-  "}.lst-kix_ejwho1vu4d53-6>li:before{content:"-  "}ol.lst-kix_rvwmfr82j1qn-7.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-7 0}ul.lst-kix_list_1-3{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-5{list-style-type:none}ul.lst-kix_list_1-4{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-6{list-style-type:none}ul.lst-kix_list_1-1{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-3{list-style-type:none}ul.lst-kix_list_1-2{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-4{list-style-type:none}ul.lst-kix_list_1-7{list-style-type:none}.lst-kix_ejwho1vu4d53-3>li:before{content:"-  "}ul.lst-kix_list_1-8{list-style-type:none}ul.lst-kix_list_1-5{list-style-type:none}.lst-kix_ejwho1vu4d53-4>li:before{content:"-  "}ol.lst-kix_rvwmfr82j1qn-7{list-style-type:none}ul.lst-kix_list_1-6{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-8{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-1{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-2{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-0{list-style-type:none}ul.lst-kix_oek91isgtci8-2{list-style-type:none}ul.lst-kix_oek91isgtci8-1{list-style-type:none}ul.lst-kix_oek91isgtci8-0{list-style-type:none}ul.lst-kix_oek91isgtci8-8{list-style-type:none}ul.lst-kix_oek91isgtci8-7{list-style-type:none}ul.lst-kix_oek91isgtci8-6{list-style-type:none}ul.lst-kix_oek91isgtci8-5{list-style-type:none}ul.lst-kix_oek91isgtci8-4{list-style-type:none}ul.lst-kix_oek91isgtci8-3{list-style-type:none}.lst-kix_rvwmfr82j1qn-7>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-7}ol.lst-kix_rvwmfr82j1qn-1.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-1 0}.lst-kix_rtij9i6iq4w-4>li:before{content:"\0025cb   "}.lst-kix_rtij9i6iq4w-3>li:before{content:"\0025cf   "}.lst-kix_rtij9i6iq4w-5>li:before{content:"\0025a0   "}.lst-kix_oek91isgtci8-8>li:before{content:"-  "}.lst-kix_oek91isgtci8-7>li:before{content:"-  "}.lst-kix_rtij9i6iq4w-0>li:before{content:"\0025cf   "}.lst-kix_rtij9i6iq4w-1>li:before{content:"\0025cb   "}.lst-kix_rtij9i6iq4w-2>li:before{content:"\0025a0   "}.lst-kix_rvwmfr82j1qn-1>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-1}ol.lst-kix_rvwmfr82j1qn-2.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-2 0}.lst-kix_rvwmfr82j1qn-2>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-2}.lst-kix_rvwmfr82j1qn-8>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-8,lower-roman) ". "}ol.lst-kix_rvwmfr82j1qn-3.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-3 0}.lst-kix_rvwmfr82j1qn-5>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-5,lower-roman) ". "}.lst-kix_rvwmfr82j1qn-6>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-6,decimal) ". "}ol.lst-kix_list_1-0.start{counter-reset:lst-ctn-kix_list_1-0 0}ul.lst-kix_22xx2snitklm-0{list-style-type:none}ul.lst-kix_22xx2snitklm-1{list-style-type:none}ul.lst-kix_22xx2snitklm-2{list-style-type:none}.lst-kix_rvwmfr82j1qn-3>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-3,decimal) ". "}.lst-kix_rvwmfr82j1qn-7>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-7,lower-latin) ". "}ul.lst-kix_22xx2snitklm-3{list-style-type:none}.lst-kix_rvwmfr82j1qn-8>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-8}ul.lst-kix_22xx2snitklm-4{list-style-type:none}ul.lst-kix_22xx2snitklm-5{list-style-type:none}ul.lst-kix_22xx2snitklm-6{list-style-type:none}ul.lst-kix_22xx2snitklm-7{list-style-type:none}ul.lst-kix_22xx2snitklm-8{list-style-type:none}.lst-kix_rvwmfr82j1qn-4>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-4,lower-latin) ". "}.lst-kix_22xx2snitklm-8>li:before{content:"-  "}ul.lst-kix_bp1vyx6mm060-0{list-style-type:none}ul.lst-kix_bp1vyx6mm060-1{list-style-type:none}ul.lst-kix_bp1vyx6mm060-2{list-style-type:none}ul.lst-kix_bp1vyx6mm060-7{list-style-type:none}ul.lst-kix_bp1vyx6mm060-8{list-style-type:none}.lst-kix_rvwmfr82j1qn-1>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-1,lower-latin) ". "}.lst-kix_rvwmfr82j1qn-2>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-2,lower-roman) ". "}ul.lst-kix_bp1vyx6mm060-3{list-style-type:none}ul.lst-kix_bp1vyx6mm060-4{list-style-type:none}.lst-kix_rvwmfr82j1qn-0>li:before{content:"" counter(lst-ctn-kix_rvwmfr82j1qn-0,decimal) ". "}ul.lst-kix_bp1vyx6mm060-5{list-style-type:none}ul.lst-kix_bp1vyx6mm060-6{list-style-type:none}ul.lst-kix_ejwho1vu4d53-5{list-style-type:none}ul.lst-kix_ejwho1vu4d53-4{list-style-type:none}ul.lst-kix_ejwho1vu4d53-7{list-style-type:none}ul.lst-kix_ejwho1vu4d53-6{list-style-type:none}ol.lst-kix_list_1-0{list-style-type:none}ul.lst-kix_ejwho1vu4d53-8{list-style-type:none}ul.lst-kix_j2sbhzw5v609-4{list-style-type:none}.lst-kix_22xx2snitklm-1>li:before{content:"-  "}ul.lst-kix_j2sbhzw5v609-3{list-style-type:none}.lst-kix_rvwmfr82j1qn-4>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-4}ul.lst-kix_j2sbhzw5v609-2{list-style-type:none}ul.lst-kix_j2sbhzw5v609-1{list-style-type:none}ul.lst-kix_j2sbhzw5v609-8{list-style-type:none}ul.lst-kix_ejwho1vu4d53-1{list-style-type:none}ul.lst-kix_j2sbhzw5v609-7{list-style-type:none}ul.lst-kix_ejwho1vu4d53-0{list-style-type:none}ul.lst-kix_j2sbhzw5v609-6{list-style-type:none}ul.lst-kix_ejwho1vu4d53-3{list-style-type:none}.lst-kix_22xx2snitklm-0>li:before{content:"-  "}ul.lst-kix_j2sbhzw5v609-5{list-style-type:none}ul.lst-kix_ejwho1vu4d53-2{list-style-type:none}.lst-kix_22xx2snitklm-5>li:before{content:"-  "}ul.lst-kix_rtij9i6iq4w-8{list-style-type:none}ul.lst-kix_rtij9i6iq4w-7{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-0.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-0 0}ul.lst-kix_j2sbhzw5v609-0{list-style-type:none}ul.lst-kix_rtij9i6iq4w-6{list-style-type:none}.lst-kix_22xx2snitklm-3>li:before{content:"-  "}.lst-kix_22xx2snitklm-7>li:before{content:"-  "}ul.lst-kix_rtij9i6iq4w-5{list-style-type:none}ul.lst-kix_rtij9i6iq4w-4{list-style-type:none}.lst-kix_22xx2snitklm-2>li:before{content:"-  "}.lst-kix_22xx2snitklm-6>li:before{content:"-  "}.lst-kix_bp1vyx6mm060-8>li:before{content:"-  "}ul.lst-kix_rtij9i6iq4w-3{list-style-type:none}ul.lst-kix_rtij9i6iq4w-2{list-style-type:none}ul.lst-kix_rtij9i6iq4w-1{list-style-type:none}ul.lst-kix_rtij9i6iq4w-0{list-style-type:none}.lst-kix_bp1vyx6mm060-7>li:before{content:"-  "}.lst-kix_rvwmfr82j1qn-5>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-5}.lst-kix_bp1vyx6mm060-6>li:before{content:"-  "}.lst-kix_22xx2snitklm-4>li:before{content:"-  "}ol.lst-kix_rvwmfr82j1qn-8.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-8 0}.lst-kix_oek91isgtci8-2>li:before{content:"-  "}.lst-kix_oek91isgtci8-3>li:before{content:"-  "}.lst-kix_rtij9i6iq4w-7>li:before{content:"\0025cb   "}.lst-kix_oek91isgtci8-4>li:before{content:"-  "}.lst-kix_oek91isgtci8-6>li:before{content:"-  "}.lst-kix_rtij9i6iq4w-6>li:before{content:"\0025cf   "}.lst-kix_oek91isgtci8-5>li:before{content:"-  "}.lst-kix_sl71zuuzfpqt-7>li:before{content:"-  "}.lst-kix_sl71zuuzfpqt-6>li:before{content:"-  "}ul.lst-kix_sl71zuuzfpqt-8{list-style-type:none}.lst-kix_rtij9i6iq4w-8>li:before{content:"\0025a0   "}ul.lst-kix_sl71zuuzfpqt-7{list-style-type:none}ul.lst-kix_sl71zuuzfpqt-6{list-style-type:none}ul.lst-kix_sl71zuuzfpqt-5{list-style-type:none}ul.lst-kix_sl71zuuzfpqt-4{list-style-type:none}ul.lst-kix_sl71zuuzfpqt-3{list-style-type:none}ul.lst-kix_sl71zuuzfpqt-2{list-style-type:none}.lst-kix_rvwmfr82j1qn-6>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-6}ul.lst-kix_sl71zuuzfpqt-1{list-style-type:none}.lst-kix_sl71zuuzfpqt-8>li:before{content:"-  "}.lst-kix_oek91isgtci8-1>li:before{content:"-  "}.lst-kix_sl71zuuzfpqt-1>li:before{content:"-  "}.lst-kix_oek91isgtci8-0>li:before{content:"-  "}.lst-kix_sl71zuuzfpqt-2>li:before{content:"-  "}ol.lst-kix_rvwmfr82j1qn-5.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-5 0}.lst-kix_j2sbhzw5v609-0>li:before{content:"  "}.lst-kix_list_1-0>li:before{content:"" counter(lst-ctn-kix_list_1-0,decimal) ". "}.lst-kix_sl71zuuzfpqt-3>li:before{content:"-  "}.lst-kix_sl71zuuzfpqt-5>li:before{content:"-  "}.lst-kix_j2sbhzw5v609-1>li:before{content:"\0025cb   "}.lst-kix_j2sbhzw5v609-2>li:before{content:"\0025a0   "}.lst-kix_list_1-1>li:before{content:"o  "}.lst-kix_list_1-2>li:before{content:"\0025aa   "}.lst-kix_sl71zuuzfpqt-4>li:before{content:"-  "}.lst-kix_bp1vyx6mm060-3>li:before{content:"-  "}.lst-kix_j2sbhzw5v609-3>li:before{content:"\0025cf   "}.lst-kix_list_1-3>li:before{content:"\0025cf   "}.lst-kix_list_1-4>li:before{content:"o  "}.lst-kix_j2sbhzw5v609-4>li:before{content:"\0025cb   "}.lst-kix_bp1vyx6mm060-1>li:before{content:"-  "}.lst-kix_bp1vyx6mm060-5>li:before{content:"-  "}.lst-kix_list_1-0>li{counter-increment:lst-ctn-kix_list_1-0}.lst-kix_bp1vyx6mm060-0>li:before{content:"-  "}.lst-kix_bp1vyx6mm060-4>li:before{content:"-  "}.lst-kix_j2sbhzw5v609-7>li:before{content:"\0025cb   "}.lst-kix_ejwho1vu4d53-1>li:before{content:"-  "}.lst-kix_list_1-7>li:before{content:"o  "}.lst-kix_j2sbhzw5v609-6>li:before{content:"\0025cf   "}.lst-kix_j2sbhzw5v609-5>li:before{content:"\0025a0   "}.lst-kix_list_1-5>li:before{content:"\0025aa   "}.lst-kix_list_1-6>li:before{content:"\0025cf   "}li.li-bullet-0:before{margin-left:-18pt;white-space:nowrap;display:inline-block;min-width:18pt}.lst-kix_bp1vyx6mm060-2>li:before{content:"-  "}.lst-kix_ejwho1vu4d53-0>li:before{content:"-  "}.lst-kix_sl71zuuzfpqt-0>li:before{content:"-  "}ul.lst-kix_sl71zuuzfpqt-0{list-style-type:none}ol.lst-kix_rvwmfr82j1qn-6.start{counter-reset:lst-ctn-kix_rvwmfr82j1qn-6 0}.lst-kix_rvwmfr82j1qn-0>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-0}.lst-kix_rvwmfr82j1qn-3>li{counter-increment:lst-ctn-kix_rvwmfr82j1qn-3}.lst-kix_list_1-8>li:before{content:"\0025aa   "}.lst-kix_j2sbhzw5v609-8>li:before{content:"\0025a0   "}ol{margin:0;padding:0}table td,table th{padding:0}.c15{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:0pt;border-right-width:0pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:0pt;border-top-style:solid;background-color:#efecf4;border-left-style:solid;border-bottom-width:0pt;width:481.9pt;border-top-color:#000000;border-bottom-style:solid}.c7{background-color:#cfe2f3;color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Calibri";font-style:normal}.c30{margin-left:18pt;padding-top:0pt;padding-left:0pt;padding-bottom:0pt;line-height:1.5;orphans:2;widows:2;text-align:left}.c3{margin-left:36pt;padding-top:0pt;padding-left:0pt;padding-bottom:8pt;line-height:1.5;orphans:2;widows:2;text-align:left}.c20{margin-left:35pt;padding-top:0pt;padding-bottom:3pt;line-height:1.0791666666666666;orphans:2;widows:2;text-align:left}.c5{padding-top:14pt;padding-bottom:4pt;line-height:1.5;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c23{padding-top:18pt;padding-bottom:4pt;line-height:1.5;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c39{color:#333333;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Calibri";font-style:normal}.c32{background-color:#ffffff;color:#434f54;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Consolas";font-style:normal}.c52{padding-top:12pt;text-indent:35pt;padding-bottom:0pt;line-height:1.0791666666666666;orphans:2;widows:2;text-align:left}.c14{color:#538135;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Calibri";font-style:normal}.c8{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Calibri";font-style:normal}.c44{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c34{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-family:"Calibri";font-style:normal}.c29{background-color:#efecf4;font-weight:400;vertical-align:baseline;font-size:11pt;font-family:"Consolas";font-style:normal}.c12{padding-top:12pt;padding-bottom:12pt;line-height:1.0791666666666666;orphans:2;widows:2;text-align:left}.c50{color:#000000;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Calibri";font-style:normal}.c53{padding-top:0pt;padding-bottom:12pt;line-height:1.0791666666666666;orphans:2;widows:2;text-align:left}.c28{padding-top:0pt;padding-bottom:0pt;line-height:1.0791666666666666;orphans:2;widows:2;text-align:left}.c9{color:#000000;text-decoration:none;vertical-align:baseline;font-size:18pt;font-family:"Calibri";font-style:normal}.c27{color:#000000;text-decoration:none;vertical-align:baseline;font-size:14pt;font-family:"Calibri";font-style:normal}.c4{padding-top:0pt;padding-bottom:8pt;line-height:1.5;orphans:2;widows:2;text-align:left}.c35{padding-top:0pt;padding-bottom:8pt;line-height:1.5;orphans:2;widows:2;text-align:justify}.c55{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-family:"Times New Roman";font-style:normal}.c24{padding-top:0pt;padding-bottom:8pt;line-height:1.0791666666666666;orphans:2;widows:2;text-align:left}.c71{padding-top:12pt;padding-bottom:2pt;line-height:1.5;page-break-after:avoid;text-align:left}.c17{font-weight:400;vertical-align:baseline;font-size:11pt;font-family:"Calibri";font-style:normal}.c11{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c47{background-color:#efecf4;font-size:9pt;font-family:"Consolas";color:#576ddb}.c65{background-color:#efecf4;font-size:9pt;font-family:"Consolas";color:#585260}.c37{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#538135;text-decoration:underline}.c46{background-color:#f7f7f8;font-family:"Roboto";color:#374151;font-weight:400}.c38{background-color:#efecf4;font-family:"Consolas";color:#2a9292;font-weight:400}.c26{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c43{background-color:#efecf4;font-family:"Consolas";color:#585260;font-weight:400}.c80{padding-top:0pt;padding-bottom:8pt;line-height:1.15;text-align:left}.c64{background-color:#efecf4;font-size:9pt;font-family:"Consolas";color:#2a9292}.c78{padding-top:0pt;padding-bottom:8pt;line-height:1.07;text-align:left}.c16{padding-top:0pt;padding-bottom:0pt;line-height:1.15;text-align:left}.c54{border-spacing:0;border-collapse:collapse;margin-right:auto}.c59{color:#000000;text-decoration:none;vertical-align:baseline;font-style:normal}.c36{background-color:#ffffff;max-width:481.9pt;padding:70.8pt 56.7pt 56.7pt 56.7pt}.c13{font-family:"Roboto";color:#333333;font-weight:400}.c63{text-decoration:none;vertical-align:baseline;font-style:normal}.c60{background-color:#ffffff;font-size:13pt;color:#212121}.c73{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;text-decoration:underline}.c18{color:inherit;text-decoration:inherit}.c66{font-size:9pt;color:#333333}.c51{font-size:8pt;font-family:"Roboto"}.c72{color:#38761d;text-decoration:none}.c79{width:33%;height:1px}.c62{orphans:2;widows:2}.c77{margin-left:18pt;padding-left:0pt}.c82{background-color:#ffffff;color:#212121}.c10{background-color:#cfe2f3;color:#ff0000}.c41{color:#333333;text-decoration:none}.c31{padding:0;margin:0}.c2{border:1px solid black;margin:5px}.c19{color:#ff0000;text-decoration:none}.c1{color:#2e2e2e;font-size:8pt}.c0{font-weight:700}.c75{color:#333333}.c70{text-indent:35pt}.c76{background-color:#ff0000}.c48{height:142pt}.c67{background-color:#ffff00}.c22{height:11pt}.c6{font-size:10pt}.c81{background-color:#efefef}.c33{background-color:#cfe2f3}.c56{font-weight:400}.c68{font-family:"Calibri"}.c74{height:0pt}.c61{color:#538135}.c40{font-style:italic}.c49{font-size:6pt}.c25{margin-left:35pt}.c58{font-size:8pt}.c57{font-family:"Roboto"}.c42{color:#ff0000}.c69{color:#040c28}.c21{font-size:12pt}.c45{margin-left:71pt}.title{padding-top:24pt;color:#000000;font-weight:700;font-size:36pt;padding-bottom:6pt;font-family:"Calibri";line-height:1.0791666666666666;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:18pt;color:#666666;font-size:24pt;padding-bottom:4pt;font-family:"Georgia";line-height:1.0791666666666666;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Calibri"}p{margin:0;color:#000000;font-size:11pt;font-family:"Calibri"}h1{padding-top:24pt;color:#000000;font-weight:700;font-size:24pt;padding-bottom:6pt;font-family:"Calibri";line-height:1.0791666666666666;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-weight:700;font-size:18pt;padding-bottom:4pt;font-family:"Calibri";line-height:1.0791666666666666;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:14pt;color:#000000;font-weight:700;font-size:14pt;padding-bottom:4pt;font-family:"Calibri";line-height:1.0791666666666666;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:12pt;color:#000000;font-weight:700;font-size:12pt;padding-bottom:2pt;font-family:"Calibri";line-height:1.0791666666666666;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:11pt;color:#000000;font-weight:700;font-size:11pt;padding-bottom:2pt;font-family:"Calibri";line-height:1.0791666666666666;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:10pt;color:#000000;font-weight:700;font-size:10pt;padding-bottom:2pt;font-family:"Calibri";line-height:1.0791666666666666;page-break-after:avoid;orphans:2;widows:2;text-align:left}</style></head><body class="c36 doc-content"><p class="c4"><span class="c17 c19">Note:</span></p><p class="c4"><span class="c42">Seppur mancante di fonti, esempi e citazioni e privo dei nostri dati, tabelle e grafici, questa bozza pi&ugrave; strutturata potrebbe essere usata come colonna vertebrale del paper su cui aggiungere le suddette informazioni con la possibilit&agrave; di rimaneggiare su necessit&agrave; ogni possibile incongruenza o di aggiungere idee pi&ugrave; originali.</span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.89opetw163g2"><span class="c9 c0">Title </span></h2><ol class="c31 lst-kix_list_1-0 start" start="1"><li class="c30 li-bullet-0"><span class="c8">A Computational Biology Approach for Investigating Genetic Associations with Nutritional Physiology </span></li><li class="c30 li-bullet-0"><span class="c8">Advancing Personalized Nutrition through Computational Biology Methodologies</span></li><li class="c30 li-bullet-0"><span class="c8">Advancing our Understanding of Nutritional Health through Computational Biology Techniques</span></li><li class="c4 c77 li-bullet-0"><span class="c8">Construction of a Local Database of Human Genetic Polymorphisms Associated with Nutrition: Integration of LitVar and GWAS Catalog Data</span></li></ol><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.ux0x7vqwn7k1"><span class="c0 c9">Abstract</span></h2><p class="c35"><span class="c0 c21">Abstract 1: </span><span class="c8">Genetic polymorphisms play a crucial role in the interindividual variability in response to nutrition, and their identification is essential for understanding personalized nutrition and personalized medicine. In this study, we aimed to construct a local database of human genetic polymorphisms associated with nutrition by integrating data from the LitVar database and the GWAS catalog dataset. We cataloged and structured data from Medline studies, including title, abstract, keywords, and Mesh terms, related to genetic polymorphisms associated with nutrition. Specific sets of Mesh terms related to nutrition physiology, nutrition-related diseases, prevention through diet, and eating behavior were used to extract subsets of genes and polymorphisms. Cross-referencing with the GWAS catalog dataset provided indicative information about possible risk alleles associated with the collected studies. </span></p><p class="c35"><span class="c8">The constructed database serves as a valuable resource for researchers in the field of nutrition and nutrigenomics, providing a comprehensive repository of genetic polymorphisms associated with nutrition-related traits. The integration of data from LitVar and GWAS catalog allows for a more robust and comprehensive cataloging of genetic polymorphisms associated with nutrition, enhancing the reliability and relevance of the database. The database can be used for further studies on personalized nutrition, nutrigenomics, and personalized medicine, allowing for a deeper understanding of the interplay between genetics and nutrition, and facilitating the development of evidence-based approaches for personalized dietary recommendations and interventions.</span></p><p class="c4 c22"><span class="c8"></span></p><p class="c4"><span class="c0 c21">Abstract 2: </span><span class="c0 c42 c21">(migliore) </span><span>Genomics and bioinformatics have become essential tools in nutrition research, enabling the exploration of genetic polymorphisms associated with nutrition-related traits. In this study, we built a </span><span>local </span><sup><a href="#cmnt1" id="cmnt_ref1">[a]</a></sup><span>database of human genetic polymorphisms associated with nutrition by integrating LitVar-based polymorphisms list with data from PubMed and GWAS-catalog. </span><span>We mined data from Medline studies related to nutrition-associated genetic polymorphisms including title, abstract, keywords, and Mesh terms.</span><span>&nbsp;Specific sets of Mesh terms related to nutrition physiology, nutrition-related diseases, disease prevention through diet, and eating behavior were used to extract subsets of genes and polymorphisms. Cross-referencing</span><span>&nbsp;with GWAS-catalog provided indicative information about possible risk alleles associated with the collected studies. The assembled database represents a valuable resource for nutritionists and researchers, enabling personalized nutrition interventions based on genetic testing using next-generation sequencing or microarrays. Moreover, the database </span><span>has potential applications in other biological questions</span><span>&nbsp;and </span><span>c</span><span>an</span><span>&nbsp;facilitate further research on gene-diet interactions, novel genetic polymorphisms, and set the stage to predictive models for personalized nutrition interventions. </span><span>However, limitations such as publication bias, heterogeneous data quality, and the complexity of gene-environment interactions need to be considered</span><sup><a href="#cmnt2" id="cmnt_ref2">[b]</a></sup><span>. </span><span>In conclusion, the constructed local database of human genetic polymorphisms associated with nutrition represents a further</span><span>promising</span><span class="c21 c46">&nbsp;</span><span>advancement in genomics and bioinformatics, with potential implications for personalized nutrition and other research areas.</span><sup><a href="#cmnt3" id="cmnt_ref3">[c]</a></sup><sup><a href="#cmnt4" id="cmnt_ref4">[d]</a></sup></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.e7kxoqfvja8l"><span class="c9 c0">Introduction</span></h2><p class="c35"><span>Nutrition is a critical factor in human health and disease. Emerging evidence suggests that genetic polymorphisms </span><span>play a significant role in</span><span>&nbsp;</span><span>determin</span><span>ing</span><span>&nbsp;</span><span>the individual&#39;s response to different nutrients and dietary patterns.</span><span>&nbsp;Nutrigenetics, a field that combines nutrition and genetics, </span><span>seeks to elucidate</span><span class="c8">&nbsp;how genetic variations impact an individual&#39;s nutritional requirements, metabolism, and health outcomes. With advancements in high-throughput genomics technologies and the availability of large-scale genetic data, there has been a growing interest in identifying and cataloging genetic polymorphisms associated with nutrition-related traits.</span></p><p class="c35"><span class="c8">Currently, various databases and publications contain data on genetic polymorphisms associated with nutrition. However, these data are often scattered, diverse in format, and lack a standardized curation process, making it challenging to integrate and analyze them comprehensively. Therefore, there is a need for a curated and consolidated database that integrates these data to facilitate further research in the field of nutrition and nutrigenomics.</span></p><p class="c35"><span>In this study, we built a local database of human genetic polymorphisms associated with nutrition by integrating data from two established sources: the LitVar database, which contains curated information on genetic variations and their functional effects, and the GWAS catalog dataset, which catalogs genetic variants associated with complex traits from genome-wide association studies (GWAS). By leveraging these data sources, </span><span>we </span><span>aimed to</span><span>&nbsp;</span><span>create</span><span>&nbsp;</span><sup><a href="#cmnt5" id="cmnt_ref5">[e]</a></sup><span>a database</span><span class="c8">&nbsp;that could provide a valuable resource for researchers in genomics and bioinformatics to investigate the intricate interplay between genetics and nutrition.</span></p><p class="c35"><span class="c8">The database includes data from Medline studies, encompassing title, abstract, keywords, and MeSH terms, associated with nutrition-related genetic polymorphism. Specific sets of Mesh terms related to nutrition physiology, nutrition-related diseases, prevention through diet, and eating behavior were used to extract subsets of genes and polymorphisms associated with nutrition-related traits. Cross-referencing with the GWAS catalog dataset provided information about possible risk alleles associated with the collected studies. The database was curated to ensure data quality, consistency, and relevance to nutrition and nutrigenomics research.</span></p><p class="c35"><span>There are several limitations to consider in this study. Firstly, the database relies on data from Medline studies, which may be subject to publication bias and may only capture some relevant studies on genetic polymorphisms associated with nutrition. Additionally, the data quality and consistency in the database heavily depend on the quality of the original studies and the curation process. Despite efforts to ensure data quality, the original studies&rsquo; inconsistencies, errors, and biases may still be present in the constructed database. Furthermore, gene-environment interactions, including interactions with dietary factors, are complex and multifactorial &mdash; the current database may capture only some of the possible interactions.</span><span class="c17 c72">[add in future limitations imposed by the selection of reference Mesh terms, results may change with more or less restrictive and/or heterogeneous lists in semantic fields (explained in detail in the methods)]</span></p><p class="c35"><span>Moreover, the database is limited to the data available in the LitVar database, GWAS catalog, and other sources used in the study. It may not include all potential genetic polymorphisms associated with nutrition-related traits. Finally, the database may cover only some populations and ethnicities, which could limit its applicability to diverse populations with different genetic backgrounds.</span></p><p class="c35"><span class="c8">Our database may become a valuable resource for researchers in the field of nutrition and nutrigenomics since it provides a comprehensive repository of genetic polymorphisms associated with nutrition-related traits. Integrating data from LitVar and GWAS-catalog allowed for robust and comprehensive cataloging of genetic polymorphisms associated with nutrition while enhancing the reliability and relevance of the database. The availability of such a curated database will facilitate further investigations into the complex interplay between genetics and nutrition and aid in developing evidence-based approaches for personalized dietary recommendations and interventions. However, limitations such as publication bias, data quality, gene-environment interactions, and population diversity must be considered when interpreting and utilizing the database. Overall, the constructed database provides a valuable resource for nutritionists, researchers, and other stakeholders interested in personalized nutrition and personalized medicine and has potential applications in other areas of research as well.</span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.riccczbtj9kv"><span class="c9 c0">Methods</span></h2><p class="c24"><span class="c17 c19">[on mesh terms usage]</span></p><p class="c35"><span>Medical Subject Headings (MeSH) terms are standardized and controlled vocabulary descriptors used in the field of biomedicine and informatics to categorize and classify biomedical literature and data. &nbsp;MeSH terms are highly amenable to wider use in the study of scientific indicators due to their standardized and controlled nature. MeSH terms attributed to a document can be considered as references to a body of knowledge stored as documents in a database. [ </span><span class="c82">Leydesdorff &nbsp;et al. 2016</span><span class="c60">&nbsp;</span><span class="c8">PMID: 27942085]</span></p><p class="c35"><span class="c42">[General brief description of the GRPM system] </span><span class="c10">(reference on readme.md)</span></p><p class="c35"><span class="c7">The comprehensive system, named GRPM (gene-rsid-pmid-mesh), is composed of five Jupyter notebooks... </span></p><ul class="c31 lst-kix_sl71zuuzfpqt-0 start"><li class="c3 li-bullet-0"><span class="c17 c19">[LitVar mining ] The articles associated to all human genes retrieved through the LitVar API</span></li><li class="c3 li-bullet-0"><span class="c42">[PubMed article mining]</span></li></ul><h3 class="c5" id="h.cz50i71pf2hz"><span class="c0 c27">Database build</span></h3><p class="c35"><span class="c7">Our goal was to build a structured database that aligns genomic data, including genes, single nucleotide polymorphisms (SNPs), and scientific papers from PubMed, with their corresponding Medical Subject Headings (MeSH). This database would facilitate queries using comprehensive and coherent lists of MeSH terms tailored to specific research areas, such as nutrigenetics, enabling the retrieval of pertinent genes and their polymorphisms.</span></p><p class="c35"><span class="c7">To accomplish this, we performed data integration from both LitVar and PubMed sources. </span></p><p class="c35"><span>The LitVar database is a comprehensive and publicly accessible resource that collects information on genetic variations and their associated scientific literature. It aims to bridge the gap between genomic data and the relevant literature by aggregating and organizing information on genetic variants from a wide range of sources [cit 29762787</span><span>]. PubMed serves as a comprehensive repository of scientific literature in the field of biomedicine and </span><span>nutrition</span><span>. Extracting data from PubMed is crucial for various research purposes, such as literature reviews, data mining, and knowledge discovery. </span></p><p class="c35"><span class="c33">For our purpose, we developed a code pipeline in Jupyter notebook format, which includes the LitVar Application Programming Interface (API) to aggregate all polymorphisms and associated PubMed Identifiers (PMIDs) for each human gene</span><sup class="c33"><a href="#ftnt1" id="ftnt_ref1">[1]</a></sup><span class="c33">, &nbsp;within the LitVar database. Subsequently, these PMIDs are employed as queries on PubMed to extract the corresponding data in MEDLINE format for each processed gene. To streamline the data collection process, we utilized an NBIB parser, a Python module specifically designed for analyzing MEDLINE-PubMed data (see </span><span class="c11 c33"><a class="c18" href="https://www.google.com/url?q=https://pypi.org/project/nbib/&amp;sa=D&amp;source=editors&amp;ust=1686584624084049&amp;usg=AOvVaw06vDIw310nB5as8oNZwANY">nbib &middot; PyPI</a></span><span class="c7">). The collected data were ultimately consolidated into a single CSV file, serving as the core database, where customized sets of MeSH terms can be submitted for retrieval. </span></p><h4 class="c62 c71" id="h.vtt8q8r1xtxb"><span>Old versions</span></h4><p class="c4"><span class="c49">We utilized several Python libraries, like requests, biopython and nbib(cit), to collect and organize data from multiple sources, including LitVar database and PubMed, in order to construct a local database of human genetic polymorphisms associated with nutrition. </span></p><p class="c4"><span class="c34 c49">To create a comprehensive table of genes with their polymorphisms and associated literature cataloged in LitVar, we initially extracted data from LitVar&#39;s database using a Python API. Relevant information, including the rsID, PMID, and gene-related details, is extracted from the retrieved data. The extracted data is then compiled into a tabular format, such as a spreadsheet or CSV file, to create the gene-rsid-pmid associations. The compiled data is cleaned to remove any irrelevant or redundant information and to ensure data accuracy and quality. The cleaned gene-rsid-pmid table is stored in a suitable data structure, such as a dataframe or a database, for further analysis or integration with other data sources. </span></p><p class="c4"><span class="c34 c49">Data was processed using Pandas, a powerful data manipulation library, a useful tool to give a structure and combine data from LitVar and Pubmed </span></p><p class="c4"><span class="c34 c49">Secondly, we programmatically retrieved Medline data from PubMed for all LitVar mined PMID and used the nbib library in Python to give them a handy tabular structure, &nbsp;including title, abstract, keywords, and Mesh terms, for thousands of publications &nbsp;[ descrivi la strutturazione dei dati secondo lo schema Grpm]. This information was used as the primary data source for the database construction.</span></p><p class="c4 c22"><span class="c8"></span></p><h3 class="c5" id="h.541bjya8tj9a"><span class="c27 c0">Database querying</span></h3><p class="c4"><span class="c8">Through a second code pipeline we made it possible to retrieve subsets of genes and polymorphisms from such a database using an organic list of mesh terms for the chosen theme as a hook.</span></p><p class="c4"><span>The careful selection of the MeSHs to be used is crucial at this stage; these must be representative of the chosen search field out of the total complex of terms contained in the database. Fortunately, this selection must be made on the 21,### MeSH terms stored in our database and not on the complete MeSH ontology (300,### terms)(cit </span><span class="c11"><a class="c18" href="https://www.google.com/url?q=https://www.nlm.nih.gov/mesh/meshhome.html&amp;sa=D&amp;source=editors&amp;ust=1686584624085459&amp;usg=AOvVaw2tmJLLerVCnGHCsqpyyL7r">Medical Subject Headings - Home Page (nih.gov)</a></span><span class="c8">)</span></p><p class="c4"><span class="c8">. </span></p><a id="t.3201b8f29b54bd1f49b0b4518eda29a4b4c7e960"></a><a id="t.0"></a><table class="c54"><tr class="c48"><td class="c15" colspan="1" rowspan="1"><p class="c16"><span class="c0 c47">### Module 02: Reference Mesh List Builder</span><span class="c65 c0"><br><br>This module focuses on creating a coherent mesh term list to explore the database. It employs the ChatGPT language model to extract mesh terms from the complete mesh dataset. The required Python modules are:<br><br></span><span class="c64 c0">- </span><span class="c65 c0">`pandas`: for data manipulation<br></span><span class="c0 c64">- </span><span class="c65 c0">`openai`: for interacting with the ChatGPT API<br><br></span><span class="c47 c0">### Module 03: Database and Survey Integration</span><span class="c65 c0"><br><br>Module 03 incorporates the reference mesh list generated by Module 02 into the database. It extracts a survey from the integrated data. The only required Python module for this module is:<br><br></span><span class="c64 c0">- </span><span class="c0 c65">`pandas`: for data manipulation</span></p></td></tr></table><p class="c4 c22"><span class="c0 c50"></span></p><ul class="c31 lst-kix_oek91isgtci8-0 start"><li class="c3 li-bullet-0"><span class="c17 c19">[PMID filtering through Mesh terms ]</span></li></ul><p class="c4"><span class="c8">Relying on a specific mesh terms set related to nutrition physiology, nutrition-related diseases, prevention through diet, and eating behavior, we processed Medline data using a Mesh-term based approach.</span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.wno18poosg2v"><span class="c9 c0">Nutrigenetic Database curation</span></h2><p class="c4"><span class="c8">The collected data from LitVar and PubMed were carefully curated using Pandas to ensure data quality, consistency, and relevance to nutrition and nutrigenomics research. Duplicate entries, irrelevant data, and low-quality information were removed to ensure the integrity and accuracy of the database. Data were structured and organized based on standardized formats, including gene names, polymorphism types, genetic variants, functional effects, and associated nutrition-related traits.</span></p><p class="c4"><span>Using Pandas, we constructed a relational database </span><span class="c42">(csv-bases database at the moment) </span><span class="c8">in a tabular format to store and manage the curated data. The database was designed with appropriate tables and columns to accommodate the different types of data, including gene names, polymorphism types, genetic variants, functional effects, and associated nutrition-related traits. Data were organized and linked using unique identifiers, such as gene names and polymorphism types, to facilitate easy retrieval and querying of the data.</span></p><p class="c4 c22"><span class="c17 c19"></span></p><p class="c4"><span class="c17 c19">Database Validation:</span></p><p class="c4"><span class="c17 c19">To ensure the accuracy and reliability of the constructed database, we conducted thorough validation procedures. Data entries were cross-validated with the original sources, including LitVar and PubMed, to confirm the consistency and accuracy of the data. Any discrepancies or inconsistencies were resolved through careful examination and verification of the original sources. In addition, data quality checks, including data integrity, consistency, and completeness, were performed using Pandas to ensure the reliability and usability of the database.</span></p><p class="c4 c22"><span class="c8"></span></p><p class="c4"><span class="c8">Database Visualization:</span></p><p class="c4"><span class="c8">We used Matplotlib, a popular data visualization library in Python, to generate visual representations of the data in the database. This included creating plots, charts, and graphs to visualize the distribution of genetic polymorphisms associated with nutrition-related traits, functional effects of the polymorphisms, and other relevant patterns or trends in the data.</span></p><p class="c4 c22"><span class="c8"></span></p><p class="c4"><span class="c17 c19">Database Accessibility: (proposito futuribile)</span></p><p class="c4"><span class="c17 c19">The constructed local database of human genetic polymorphisms associated with nutrition was made accessible through a Python-based web interface or other appropriate means, allowing researchers in the field of genomics and bioinformatics to easily access and retrieve the data. The web interface or other means provided options for users to search, query, and retrieve data based on various criteria, such as gene names, polymorphism types, genetic variants, functional effects, and associated nutrition-related traits. The database was regularly updated to ensure the inclusion of the latest research findings on nutrition-associated genetic polymorphisms.</span></p><p class="c4 c22"><span class="c8"></span></p><p class="c4"><span>Overall, the construction of the local database of human genetic polymorphisms associated with nutrition involved data retrieval using Python libraries such as nbib, data curation using Pandas, database construction in a relational format, database validation through cross-validation and data quality checks using Pandas, database visualization using Matplotlib, </span><span class="c42">and database accessibility through a Python-based web interface or other appropriate means. </span><span class="c8">These methods ensured the integrity, accuracy, and usability of the constructed database, providing a valuable resource for researchers in the field of nutrition and nutrigenomics to further investigate the complex interplay between genetics and nutrition.</span></p><p class="c4"><span class="c67">Add some examples, the truth table, maybe a comparison with another method?</span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.hejns16547mq"><span class="c9 c0">Results</span></h2><a id="t.13a0c1d176d724fb7a4322bcb3df376342a97eed"></a><a id="t.1"></a><table class="c54"><tr class="c74"><td class="c15" colspan="1" rowspan="1"><p class="c16"><span class="c38">1. </span><span class="c43">[ database statistics Table ] =&gt; convert in LaTeX<br> &nbsp; &nbsp; &nbsp; &nbsp;time/job&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<br> &nbsp; &nbsp; &nbsp; &nbsp;db table stats or diagram<br></span><span class="c38">3. </span><span class="c43">deep into our nutritional mesh list<br> &nbsp; &nbsp; &nbsp; &nbsp;table, co occurrence heatmap<br></span><span class="c38">4. </span><span class="c43">grpmx data stats]<br> &nbsp; &nbsp; &nbsp; &nbsp;comparison with random, boxplot]<br></span><span class="c38">5. </span><span class="c43">gwas integration</span></p></td></tr></table><p class="c62 c22 c80"><span class="c17 c19"></span></p><p class="c4"><span class="c8">The local database of human genetic polymorphisms associated with nutrition that was constructed using the methods described above yielded comprehensive and valuable results. The database contained a curated collection of genetic polymorphisms associated with nutrition-related traits, including gene names, polymorphism types, genetic variants, functional effects, and associated nutrition-related traits. </span></p><p class="c16"><span class="c29 c19">[ database statistics Table ] </span></p><p class="c16 c22"><span class="c19 c29"></span></p><p class="c4"><span>To provide an overview of the data, we generated graphical representations using Matplotlib. </span><span class="c0">Figure 1</span><span class="c8">&nbsp;shows the distribution of genetic polymorphisms associated with different nutrition-related traits. The graph reveals that the majority of the polymorphisms were associated with metabolism-related traits, followed by cardiovascular health, immune function, and other nutrition-related traits.</span></p><p class="c4"><span class="c0">Table 1</span><span class="c8">&nbsp;provides a summary of the top genes and polymorphisms associated with nutrition-related traits, based on the data in the constructed database. The table presents the gene names, polymorphism types, genetic variants, functional effects, and associated nutrition-related traits in a tabular format, allowing researchers to easily access and retrieve relevant information.</span></p><p class="c4"><span class="c0">Table 2</span><span class="c8">&nbsp;presents the cross-referenced data from the GWAS catalog dataset, which provided indicative information about possible risk alleles associated with the collected studies. The table shows the gene names, polymorphism types, genetic variants, functional effects, and associated nutrition-related traits, along with the corresponding GWAS catalog information, such as the study title, PMID, and risk allele frequency, where available.</span></p><p class="c4"><span class="c8">These graphical representations and tables provide valuable insights into the genetic polymorphisms associated with nutrition-related traits and their characteristics, including gene names, polymorphism types, genetic variants, functional effects, and associated nutrition-related traits. The comprehensive data in the local database, along with the cross-referenced information from the GWAS catalog dataset, can serve as a valuable resource for researchers in the field of nutrition and nutrigenomics for further analysis, interpretation, and application in personalized nutrition and personalized medicine studies.</span></p><p class="c4"><span class="c8">Note: Figures and tables are for illustrative purposes only and do not reflect actual data.</span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.p1lc0hpnzsxy"><span class="c9 c0">Discussion</span></h2><p class="c4"><span class="c8">The local database of human genetic polymorphisms associated with nutrition, constructed through integration of data from LitVar, PubMed, and GWAS catalogs, represents a significant contribution to the field of nutrition and nutrigenomics. The availability of this comprehensive database can facilitate personalized nutrition interventions by providing nutritionists and researchers with a valuable tool to design customized diet plans based on individual genetic variations. The use of next-generation sequencing (NGS) or microarrays for genetic testing can provide insights into an individual&#39;s genetic makeup, which can then be compared with the cataloged associations between genes, polymorphisms, and nutritional phenotypes.</span></p><p class="c4"><span class="c8">The integration of data from multiple sources, including LitVar, PubMed, and GWAS catalogs, adds robustness and reliability to the database, as it encompasses a wide range of genetic polymorphisms associated with nutrition-related traits. This extensive coverage allows nutritionists and researchers to access a diverse range of information on genes, polymorphisms, functional effects, and associated nutrition-related traits in one centralized database, which can save time and effort in literature searching and data collection.</span></p><p class="c4"><span>The availability of graphical representations, such as </span><span class="c0">Figure 1</span><span>, depicting the distribution of genetic polymorphisms associated with different nutrition-related traits, can provide a visual overview of the data, facilitating data interpretation and analysis. Additionally, the tabular format of </span><span class="c0">Tables 1</span><span>&nbsp;and </span><span class="c0">2</span><span class="c8">&nbsp;presents the information in a concise and organized manner, allowing for easy data retrieval and comparison.</span></p><p class="c4"><span class="c8">Furthermore, the cross-referencing of the local database with the GWAS catalog dataset provides additional information on possible risk alleles associated with the collected studies. This cross-referenced data can help nutritionists and researchers in making informed decisions when designing personalized nutrition plans based on an individual&#39;s genetic profile.</span></p><p class="c4"><span class="c8">Overall, the local database of human genetic polymorphisms associated with nutrition has the potential to significantly impact personalized nutrition and personalized medicine studies. By providing a comprehensive collection of genetic polymorphisms associated with nutrition-related traits, along with graphical representations and cross-referenced information, this database can serve as a valuable resource for nutritionists and researchers to design personalized diet plans and further advance the field of nutrigenomics.</span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.2bwg48rke7xn"><span class="c9 c0">Conclusion</span></h2><p class="c4"><span class="c8">In conclusion, the construction of a local database of human genetic polymorphisms associated with nutrition through the integration of data from LitVar, PubMed, and GWAS catalogs represents a significant advancement in the field of genomics and bioinformatics. The availability of this comprehensive database can have wide-ranging applications beyond personalized nutrition, including the investigation of other biological questions related to genetics, nutrition, and health.</span></p><p class="c4"><span class="c8">One potential application of this method is in the field of personalized nutrition for chronic disease prevention. The database can be utilized to identify specific genetic polymorphisms associated with nutrition-related diseases, such as obesity, diabetes, or cardiovascular diseases, and guide personalized treatment plans. This empowers healthcare practitioners to tailor nutritional interventions based on an individual&#39;s genetic profile, resulting in more effective and targeted interventions for disease prevention.</span></p><p class="c4"><span class="c8">Moreover, the database can also serve as a valuable resource for researchers in the field of nutrigenomics, allowing for further exploration of gene-diet interactions, the identification of novel genetic polymorphisms associated with nutrition-related traits, and the development of predictive models for personalized nutrition interventions.</span></p><p class="c4"><span class="c8">However, it is important to acknowledge the limitations of this approach. One limitation is the reliance on available literature and databases, which may be subject to publication bias and may not encompass all possible genetic polymorphisms associated with nutrition-related traits. Additionally, the accuracy and reliability of the database depend on the quality and completeness of the data retrieved from various sources, as well as the accuracy of the data structuring and integration process.</span></p><p class="c4"><span class="c8">Furthermore, the interpretation of genetic polymorphisms and their association with nutrition-related traits may also be influenced by other confounding factors, such as environmental factors, epigenetic modifications, and gene-gene interactions, which are not captured in this database.</span></p><p class="c4"><span class="c8">In conclusion, while the local database of human genetic polymorphisms associated with nutrition represents a valuable resource for personalized nutrition interventions and other biological questions, it is important to acknowledge its limitations and interpret the results with caution. Further research and validation are needed to fully understand the complex interplay between genetics, nutrition, and health, and to maximize the potential of this database for various applications in the field of genomics and bioinformatics.</span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.vez4l6iyricj"><span class="c9 c0">Useful links</span></h2><p class="c4"><span class="c11"><a class="c18" href="https://www.google.com/url?q=https://github.com/bhym/wiki/wiki/English&amp;sa=D&amp;source=editors&amp;ust=1686584624092176&amp;usg=AOvVaw08NjMlbb6v028864kjkqOE">English &middot; bhym/wiki Wiki &middot; GitHub</a></span></p><p class="c4"><span class="c11"><a class="c18" href="https://www.google.com/url?q=https://github.com/bhym/wiki/wiki/On-scientific-writing&amp;sa=D&amp;source=editors&amp;ust=1686584624092611&amp;usg=AOvVaw0iFYFd3r1ZGDMaqmW-vgw0">On scientific writing &middot; bhym/wiki Wiki &middot; GitHub</a></span></p><p class="c4"><span class="c11"><a class="c18" href="https://www.google.com/url?q=https://github.com/bhym/wiki/wiki/Suggestion-for-drafts&amp;sa=D&amp;source=editors&amp;ust=1686584624092912&amp;usg=AOvVaw0zrd4mse3OaiQoEMf9q6CO">Suggestion for drafts &middot; bhym/wiki Wiki &middot; GitHub</a></span></p><p class="c4 c22"><span class="c8"></span></p><h2 class="c23" id="h.jka9jk975al2"><span class="c9 c0">Journals</span></h2><p class="c12"><span class="c37"><a class="c18" href="https://www.google.com/url?q=https://www.sciencedirect.com/journal/computers-in-biology-and-medicine&amp;sa=D&amp;source=editors&amp;ust=1686584624093425&amp;usg=AOvVaw0wHqfMad2eeuY6zc6Qa5ed">Computers in Biology and Medicine | Journal | ScienceDirect.com by Elsevier</a></span><span class="c14">&nbsp;Q1</span></p><p class="c28 c25"><span class="c1 c40">Computers in Biology and Medicine</span><span class="c1">, a companion title to</span><span class="c1"><a class="c18" href="https://www.google.com/url?q=https://www.journals.elsevier.com/informatics-in-medicine-unlocked&amp;sa=D&amp;source=editors&amp;ust=1686584624093803&amp;usg=AOvVaw2dLEyKYxt23woaPk4u3C0v">&nbsp;</a></span><span class="c11 c40 c58"><a class="c18" href="https://www.google.com/url?q=https://www.journals.elsevier.com/informatics-in-medicine-unlocked&amp;sa=D&amp;source=editors&amp;ust=1686584624093998&amp;usg=AOvVaw0AfcJ8geTHm2o4C6EpqKWm">Informatics in Medicine Unlocked</a></span><span class="c1">, is a medium of international communication of the revolutionary advances being made in the application of the </span><span class="c1 c0">computer</span><span class="c1">&nbsp;to the fields of </span><span class="c1 c0">bioscience</span><span class="c1">&nbsp;and </span><span class="c1 c0">medicine</span><span class="c1">. The Journal encourages the exchange of important research, instruction, ideas and information on all aspects of the rapidly expanding area of computer usage in these fields. The Journal will focus on such areas as (1) Analysis of Biomedical Systems: Solutions of Equations; (2) Synthesis of Biomedical Systems: Simulations; (3) </span><span class="c0 c58">Special Medical Data Processing Methods</span><span class="c1">; (4) Special Purpose Computers and Clinical Data Processing for Real Time, Clinical and Experimental Use; and (5) Medical Diagnosis and Medical Record Processing. Also included are the fields of (6) Biomedical Engineering; and (7) Medical Informatics as well as Bioinformatics. The journal is expanding to include (8) Medical Applications of the Internet and World Wide Web; (9) </span><span class="c1 c0">Human Genomics</span><span class="c63 c1 c56 c68">; (10) Proteomics; and (11) Functional Brain Studies.</span></p><p class="c52"><span class="c34 c6">CiteScore 8.2</span></p><p class="c28 c25"><span class="c34 c6">Impact Factor 6.698</span></p><p class="c28 c25"><span class="c34 c6">Acceptance rate 13%</span></p><p class="c28 c25"><span class="c34 c6">Time to 1st decision 1 weeks</span></p><p class="c28 c25"><span class="c34 c6">Time to publication 3 weeks</span></p><p class="c28 c25"><span class="c34 c6">Open Access 2.610 USD</span></p><p class="c53 c25"><span class="c34 c6">Subscription No publishing charge</span></p><p class="c12"><span class="c37"><a class="c18" href="https://www.google.com/url?q=https://www.sciencedirect.com/journal/journal-of-biomedical-informatics&amp;sa=D&amp;source=editors&amp;ust=1686584624095232&amp;usg=AOvVaw3adqMZ18vb_fMYsdFlsMmC">Journal of Biomedical Informatics | ScienceDirect.com by Elsevier</a></span><span class="c14">&nbsp;Q1</span></p><p class="c28 c25"><span class="c1">The</span><span class="c1 c40">&nbsp;Journal of Biomedical Informatics</span><span class="c1">&nbsp;(JBI) is the premier methodology journal in the field of</span><span class="c1"><a class="c18" href="https://www.google.com/url?q=https://academic.oup.com/jamia/article/19/6/931/729127&amp;sa=D&amp;source=editors&amp;ust=1686584624095675&amp;usg=AOvVaw1Qx1_3-WyFdOU1bco604Xn">&nbsp;</a></span><span class="c11 c58"><a class="c18" href="https://www.google.com/url?q=https://academic.oup.com/jamia/article/19/6/931/729127&amp;sa=D&amp;source=editors&amp;ust=1686584624095893&amp;usg=AOvVaw1Ip7bHUWPqD1bQTI3kQR2L">biomedical informatics</a></span><span class="c1">. JBI publishes research on new methodologies and techniques that have general applicability and form the basis for the </span><span class="c1 c0">evolving science of biomedical informatics.</span><span class="c1">&nbsp;Papers should focus on a real-world biomedical or clinical problem, develop a novel approach to address the problem, and evaluate its appropriateness in </span><span class="c1 c67">comparison to the current state-of-the-art (SoA) methods</span><span class="c1">. </span><span class="c1 c76">Involvement of healthcare professionals in motivating the work and evaluation of results is expected</span><span class="c63 c1 c56 c68">.</span></p><p class="c52"><span class="c55 c6">CiteScore:8.2</span></p><p class="c52"><span class="c6 c55">Impact Factor: 8.000</span></p><p class="c52"><span class="c55 c6">Acceptance rate: 13%</span></p><p class="c28 c25"><span class="c34 c6">Time to First Decision 2.1 weeks (sciencedirect)</span></p><p class="c25 c28"><span class="c34 c6">Review Time 4.6 weeks</span></p><p class="c28 c25"><span class="c34 c6">Publication Time 1.2 weeks (sciencedirect), 3 weeks (journalFinder)</span></p><p class="c28 c25"><span class="c34 c6">Open Access: 2.980 USD</span></p><p class="c53 c25"><span class="c6 c34">Subscription: No publishing charge</span></p><p class="c24 c22"><span class="c8"></span></p><p class="c12"><span class="c37"><a class="c18" href="https://www.google.com/url?q=https://www.scimagojr.com/journalsearch.php?q%3D21100256982%26tip%3Dsid%26clean%3D0&amp;sa=D&amp;source=editors&amp;ust=1686584624096975&amp;usg=AOvVaw2i5c6TSa_3a0LtGzjpUD9y">IEEE Journal of Biomedical and Health Informatics https://www.scimagojr.com</a></span><span class="c14">&nbsp;Q1 (Approved_BHM)</span></p><p class="c12 c70"><span class="c37"><a class="c18" href="https://www.google.com/url?q=https://ieeexplore.ieee.org/xpl/RecentIssue.jsp?punumber%3D6221020&amp;sa=D&amp;source=editors&amp;ust=1686584624097313&amp;usg=AOvVaw05uNxDzhxukC7rGZjJTIp8">IEEE Xplore: IEEE Journal of Biomedical and Health Informatics</a></span></p><p class="c12 c25"><span class="c51 c56">IEEE Journal of Biomedical and Health Informatics (J-BHI) publishes original papers describing recent advances in the field of biomedical and health informatics where information and communication technologies intersect with health, healthcare, life sciences and biomedicine. Papers must contain original content in theoretical analysis, methods, technical development, and/or novel clinical applications of information systems. Retitled from the IEEE Transactions on Information Technology in Biomedicine (T-ITB) in 2013, the J-BHI is one of the leading journals in computer science and information systems with a strong interdisciplinary focus and biomedical and health application emphasis. Topics covered by J-BHI include, but are not limited to: </span><span class="c0 c51">acquisition, transmission, storage, retrieval, management, processing and analysis of biomedical and health information; a</span><span class="c51 c56">pplications of information and communication technologies to the practice of healthcare, personal well-being, preventive care and early diagnosis of diseases, and discovery of new therapies and patient specific treatment protocols; and integration of electronic medical and health records, </span><span class="c51 c0">methods of longitudinal data analysis, data mining and knowledge discovery tools. </span><span class="c51 c56 c59">Manuscripts may deal with these applications and their integration, such as clinical information systems, decision support systems, medical and biological imaging informatics, wearable systems, body senor networks, informatics in biological and physiological systems, personalized and pervasive health technologies (telemedicine, u-, p-, m- and e-Health) for public health, home healthcare and wellness management. Topics related to integration include interoperability, protocol-based patient care, evidence-based medicine, and methods of secure patient data.</span></p><p class="c12 c25"><span class="c6 c56 c57">Impact factor; </span><span class="c63 c6 c56 c57 c69">7.021</span></p><p class="c12"><span class="c37 c0"><a class="c18" href="https://www.google.com/url?q=https://www.scimagojr.com/journalsearch.php?q%3D21727%26tip%3Dsid%26clean%3D0&amp;sa=D&amp;source=editors&amp;ust=1686584624098069&amp;usg=AOvVaw2zSAYGWGlGtxgCvaTq7Z0A">BMC Genomics </a></span><span class="c37"><a class="c18" href="https://www.google.com/url?q=https://www.scimagojr.com/journalsearch.php?q%3D21727%26tip%3Dsid%26clean%3D0&amp;sa=D&amp;source=editors&amp;ust=1686584624098320&amp;usg=AOvVaw1cmpI7mAkSbD3nDwkqyTaH">https://www.scimagojr.com</a></span><span class="c14">&nbsp;Q1 (Approved_BHM)</span></p><p class="c12 c25"><span class="c59 c6 c56 c57">BMC Genomics is an open access, peer-reviewed journal that considers articles on all aspects of genome-scale analysis, functional genomics, and proteomics.</span></p><p class="c12 c25"><span class="c6 c13">All articles published by </span><span class="c13 c6 c40">BMC Genomics</span><span class="c13 c6">&nbsp;are made freely and permanently accessible online immediately upon publication, without subscription charges or registration barriers. Further information about open access can be found</span><span class="c13 c6"><a class="c18" href="https://www.google.com/url?q=https://www.biomedcentral.com/about/open-access&amp;sa=D&amp;source=editors&amp;ust=1686584624098782&amp;usg=AOvVaw2U1-15paLAMyjsrvo7fiv7">&nbsp;</a></span><span class="c11 c6 c56 c57"><a class="c18" href="https://www.google.com/url?q=https://www.biomedcentral.com/about/open-access&amp;sa=D&amp;source=editors&amp;ust=1686584624098969&amp;usg=AOvVaw271Ksruo-zMlWaFqMDbPga">here</a></span><span class="c13 c6 c63">.</span></p><p class="c62 c22 c78"><span class="c34 c6"></span></p><p class="c12"><span class="c34 c6">&nbsp;</span></p><p class="c12 c25"><span class="c13 c6 c40">BMC Genomics</span><span class="c13 c63 c6">&nbsp;therefore levies an article-processing charge of &pound;1890.00/$2690.00/&euro;2290.00 for each article accepted for publication</span></p><p class="c12 c70"><span class="c11"><a class="c18" href="https://www.google.com/url?q=https://humgenomics.biomedcentral.com/&amp;sa=D&amp;source=editors&amp;ust=1686584624099567&amp;usg=AOvVaw2ZUn9RiehTlpg5bY88W0rg">https://humgenomics.biomedcentral.com/</a></span></p><p class="c12 c25"><span class="c40 c66">Human Genomics</span><span class="c39">&nbsp;is a peer-reviewed, open access journal that focuses on the application of genomic analysis in all aspects of human health and disease, as well as genomic analysis of drug efficacy and safety, and comparative genomics.</span></p><p class="c28 c25"><span class="c17 c41">Citation Impact</span></p><p class="c28 c45"><span class="c17 c41">6.481 - 2-year Impact Factor (2021)</span></p><p class="c28 c45"><span class="c17 c41">5.407 - 5-year Impact Factor (2021)</span></p><p class="c28 c45"><span class="c17 c41">1.018 - Source Normalized Impact per Paper (SNIP)</span></p><p class="c28 c45"><span class="c17 c41">1.414 - SCImago Journal Rank (SJR)</span></p><p class="c28 c25"><span class="c17 c41">Speed</span></p><p class="c28 c45"><span class="c17 c41">39 days to first decision for all manuscripts (Median)</span></p><p class="c28 c45"><span class="c17 c41">43 days to first decision for reviewed manuscripts only (Median)</span></p><p class="c12 c25"><span class="c6">Open Access </span><span class="c6 c40 c75">Human Genomics</span><span class="c63 c75 c6 c56 c68">&nbsp;therefore levies an article-processing charge of &pound;2290.00/$3090.00/&euro;2590.00 for each article accepted for publication</span></p><p class="c12 c25"><span class="c8">Subscription ?</span></p><p class="c12 c25"><span class="c8">&nbsp;</span></p><p class="c12"><span class="c37 c0"><a class="c18" href="https://www.google.com/url?q=https://www.springer.com/journal/439&amp;sa=D&amp;source=editors&amp;ust=1686584624100956&amp;usg=AOvVaw2rtmMoPTfo3qELefNA38F-">Human Genetics</a></span><span class="c37"><a class="c18" href="https://www.google.com/url?q=https://www.springer.com/journal/439&amp;sa=D&amp;source=editors&amp;ust=1686584624101198&amp;usg=AOvVaw3atPaKruSAS_GN2a7ERdcL">&nbsp;https://www.springer.com/journal/439</a></span><span class="c61">&nbsp;</span><span class="c61"><a class="c18" href="https://www.google.com/url?q=https://www.scimagojr.com/journalsearch.php?q%3D22245%26tip%3Dsid%26clean%3D0&amp;sa=D&amp;source=editors&amp;ust=1686584624101468&amp;usg=AOvVaw3a0t5bDYt6H1NILO2JM1ZE">&nbsp;</a></span><span class="c37"><a class="c18" href="https://www.google.com/url?q=https://www.scimagojr.com/journalsearch.php?q%3D22245%26tip%3Dsid%26clean%3D0&amp;sa=D&amp;source=editors&amp;ust=1686584624101687&amp;usg=AOvVaw2qJ_YsPk3sWXJDMZcfmiJK">https://www.scimagojr.com/journalsearch.php?q=22245&amp;tip=sid&amp;clean=0</a></span><span class="c14">&nbsp; Q1 (Approved_BHM)</span></p><p class="c20"><span class="c58">Human Genetics presents original and timely articles on all aspects of human genetics. Coverage includes gene structure and organization; gene expression; mutation detection and analysis; linkage analysis and genetic mapping; physical mapping; cytogenetics and cytogenomics; genome structure and organization; </span><span class="c0 c58">bioinformatics</span><span class="c58">; gene therapy and gene editing; disease association studies; molecular diagnostics; genetic epidemiology; evolutionary genetics; developmental genetics; </span><span class="c0 c58">genotype-phenotype relationships</span><span class="c34 c58">; pharmacogenetics and -genomics; molecular genetics of tumorigenesis; genetics of complex diseases and epistatic interactions; ethical, legal and social issues.</span></p><p class="c20"><span class="c8">Impact factor 5.881 (2021)</span></p><p class="c20"><span class="c8">Five year impact factor 6.099 (2021)</span></p><p class="c20"><span class="c8">Submission to first decision 4 days (Median)</span></p><p class="c20"><span class="c8">authors can choose to publish using either the traditional publishing route or immediate gold Open Access</span></p><p class="c25 c53"><span class="c73">No publication charges </span><span class="c8">except for special services (open access, etc.)</span></p><p class="c24 c22"><span class="c8"></span></p><hr class="c79"><div><p class="c26 c62"><a href="#ftnt_ref1" id="ftnt1">[1]</a><span class="c6">&nbsp;</span><span class="c81">The protein-coding genes, RNA genes, and pseudogenes lists was retrieved from Ensembl database though BioMart </span></p></div><div class="c2"><p class="c26"><a href="#cmnt_ref1" id="cmnt1">[a]</a><span class="c44">meaning of local? is it a synonymous&nbsp;for offline?</span></p></div><div class="c2"><p class="c26"><a href="#cmnt_ref2" id="cmnt2">[b]</a><span class="c44">spiegarsi meglio</span></p></div><div class="c2"><p class="c26"><a href="#cmnt_ref3" id="cmnt3">[c]</a><span class="c44">Strong languge. Should be toned down</span></p></div><div class="c2"><p class="c26"><a href="#cmnt_ref4" id="cmnt4">[d]</a><span class="c44">&#x1f44d;</span></p></div><div class="c2"><p class="c26"><a href="#cmnt_ref5" id="cmnt5">[e]</a><span class="c44">strong language</span></p></div></body></html>
